### PR TITLE
Add 0.4.0 version

### PR DESCRIPTION
**Pull request by nawordar**

### CODE REVIEW

**Comment by nawordar** [comment]
Sorry, I don't get how to make that into one commit

**Comment by nawordar** [comment]
OK I think I made it

File: upstream.wrap [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[wrap-file]
+directory = wren-0.4.0
+
+source_url = https://github.com/wren-lang/wren/archive/0.4.0.tar.gz
+source_filename = 0.4.0.tar.gz
+source_hash = 23c0ddeb6c67a4ed9285bded49f7c91714922c2e7bb88f42428386bf1cf7b339
```
Comment (jpakkane) [inline upstream.wrap:6]: Nowadays all wrap files should have a `provide` section:

https://mesonbuild.com/Wrap-dependency-system-manual.html#provide-section

Comment (nawordar) [reply upstream.wrap:6]: I've added the `[provide]` section. Could you look at it?

**Pull request merged**